src/test/javascript/spec/component/learning-paths/pages/learning-path-student-page.component.spec.ts

```diff
@@ -111,4 +112,34 @@ describe('LearningPathStudentPageComponent', () => {
         expect(loadingSpy).toHaveBeenCalledWith(true);
         expect(loadingSpy).toHaveBeenCalledWith(false);
     });
+
+    it('should generate learning path on click', async () => {
+        jest.spyOn(learningPathApiService, 'getLearningPathId').mockRejectedValue(new EntityNotFoundError());
+        const generateLearningPathSpy = jest.spyOn(learningPathApiService, 'generateLearningPath').mockResolvedValue(learningPathId);
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+
+        const generateLearningPathButton = fixture.debugElement.query(By.css('#generate-learning-path-button'));
+        generateLearningPathButton.nativeElement.click();
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(component.learningPathId()).toEqual(learningPathId);
+        expect(generateLearningPathSpy).toHaveBeenCalledWith(courseId);
+    });
+
+    it('should set isLoading correctly during learning path generation', async () => {
+        jest.spyOn(learningPathApiService, 'getLearningPathId').mockRejectedValue(new EntityNotFoundError());
+        jest.spyOn(learningPathApiService, 'generateLearningPath').mockResolvedValue(learningPathId);
+        const loadingSpy = jest.spyOn(component.isLoading, 'set');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(loadingSpy).toHaveBeenCalledWith(true);
+        expect(loadingSpy).toHaveBeenCalledWith(false);
```

Comment:
Just a thought: I wonder if we could be more specific here. Would `toHaveBeenNthCalledWith(1, true)` and `toHaveBeenNthCalledWith(2, false)` work? Same for the test `should set isLoading correctly during learning path load`

This way, we'd be making sure that the order of values is also correct